pkgs/stdenv/generic/setup.sh

```diff
@@ -1057,7 +1057,11 @@ _defaultUnpack() {
         case "$fn" in
             *.tar.xz | *.tar.lzma | *.txz)
                 # Don't rely on tar knowing about .xz.
-                xz -d < "$fn" | tar xf - --warning=no-timestamp
+                # Additionally, we have multiple different xz binaries with different feature sets in different
+                # stages. The XZ_OPT env var is only used by the full "XZ utils" implementation, which supports
+                # the --threads (-T) flag. This allows us to enable multithreaded decompression exclusively on
+                # that implementation, without the use of complex bash conditionals and checks.
+                XZ_OPT="--threads=$NIX_BUILD_CORES" xz -d < "$fn" | tar xf - --warning=no-timestamp
```

Comment:
Should this be dependent upon `enableParallelBuilding` (or maybe a new `enableParallelUnpacking`) being enabled, like in `buildPhase()`?

https://github.com/NixOS/nixpkgs/blob/570757826e62bcc814068e233e71eeb2f4294f50/pkgs/stdenv/generic/setup.sh#L1289-L1293

Reply:
As far as I am aware, `enableParallelBuilding` serves a different purpose.

E.g. certain `Makefile`s won't work with `-j${NIX_BUILD_CORES}`, which is why it is still not enabled by default (though there have been attempts at changing the default for `mkDerivation` in the past).
See #142338 and https://discourse.nixos.org/t/rfc-make-stdenv-to-build-in-parallel-by-default/15684.

`xz`, on the other hand, produces the same `tar` output both in multi-threaded and single-threaded mode. So there shouldn't be a single breakage due to this change.